registry/github/parse.go

```diff
@@ -12,8 +12,11 @@ import (
 	"github.com/goware/urlx"
 )
 
-// Parse creates the registry source object from a template path.
-func (c *client) Parse(path string) (*registry.Source, error) {
+const defaultRef = "main"
```

Comment:
Are we sure we want to make `main` the fall back branch?

I understand why the change is being made from `master` to `main` 👍 

But my concern is for all the repos that are using templates, but are still using `master` as their default branch.

What happens in that scenario?

Reply:
I think it's fine. Default branch passed in from the context should always be either `master` (current default if nothing was passed in at activation) or something the user defined (if the repo was activated via CLI or API; UI doesn't allow customization there). From what I can tell, we currently do not allow the user to activate a repo with an empty default branch, so the likelyhood of this falling back to `main` are basically nil, as far as I can tell. But it's possible I miss(ed|-understood) something.

If somehow it got this state, the template would fail to be pulled just as if someone's default is not `master` and they were trying to reference a template without `@<branch>`. I'm fine with that, personally.

However, I think it might be good to modify the docs to encourage the use of `@<branch>` identifier when referencing templates as a general good practice to be explicit.

As an aside, I do have another change in the hopper for the API to deal with https://github.com/go-vela/server/blob/master/api/repo.go#L113 and https://github.com/go-vela/server/blob/master/api/repo.go#L183

Reply:
Sure 👍 

I'm still a little uncomfortable with the change (setting the fallback default to `main`) because of the potential impact of breaking pipelines.

But perhaps I'm being overly cautious and this is a "rip the bandaid" scenario?

At any rate, if others don't feel strongly then I don't want to be the stick in the mud.

Reply:
Thanks for raising the concern. I do think it's a "rip the bandaid" scenario, but I also think the risk is very low. I can be convinced otherwise if there's something I didn't think of or code I missed somewhere. Regardless, thanks for checking in out.